airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -170,74 +165,140 @@ class Metafields(IncrementalShopifyStream):
         order_id = stream_slice["order_id"]
         return f"orders/{order_id}/{self.data_field}.json"
 
-    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+    def read_records(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {}
         orders_stream = Orders(authenticator=self.authenticator, shop=self.shop, start_date=self.start_date, api_password=self.api_password)
         for data in orders_stream.read_records(sync_mode=SyncMode.full_refresh):
-            yield from super().read_records(stream_slice={"order_id": data["id"]}, **kwargs)
+            refund_records = super().read_records(stream_slice={"order_id": data["id"]}, **kwargs)
+            # Getting records >= state
+            if stream_state:
+                # For the case where the order has multiple refunds
+                for record in refund_records:
+                    if record[self.cursor_field] >= stream_state.get(self.cursor_field):
+                        yield record
+            else:
+                yield from refund_records
 
 
 class OrderRisks(IncrementalShopifyStream):
     data_field = "risks"
+    cursor_field = "id"
 
     def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
         order_id = stream_slice["order_id"]
         return f"orders/{order_id}/{self.data_field}.json"
 
-    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {self.cursor_field: max(latest_record.get(self.cursor_field, 0), current_stream_state.get(self.cursor_field, 0))}
+
+    def read_records(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {}
         orders_stream = Orders(authenticator=self.authenticator, shop=self.shop, start_date=self.start_date, api_password=self.api_password)
         for data in orders_stream.read_records(sync_mode=SyncMode.full_refresh):
-            yield from super().read_records(stream_slice={"order_id": data["id"]}, **kwargs)
+            refund_records = super().read_records(stream_slice={"order_id": data["id"]}, **kwargs)
+            # Getting records >= state
+            if stream_state:
+                # For the case where the order has multiple risks
+                for record in refund_records:
+                    if record[self.cursor_field] >= stream_state.get(self.cursor_field):
+                        yield record
+            else:
+                yield from refund_records
```

Comment:
I see similar code in multiple places. Could you create a create a function and call it in multiple spots ?

Reply:
Done.